packages/runtime-common/realm.ts

```diff
@@ -1251,7 +1251,11 @@ export class Realm {
     let resources = [primaryResource, ...included];
     let primaryResourceURL: URL | undefined;
     for (let [i, resource] of resources.entries()) {
-      if (i > 0 && typeof resource.lid !== 'string') {
+      if (
+        (i > 0 && typeof resource.lid !== 'string') ||
+        // TODO test this
+        (resource.meta.realmURL && resource.meta.realmURL !== this.url)
+      ) {
```

Comment:
TODO test thru "lid not for you" scneario. assert that relationship remains null